src/main/java/io/openliberty/tools/common/plugins/util/DevUtil.java

```diff
@@ -1416,9 +1421,14 @@ protected void recompileJava(Collection<File> javaFilesChanged, List<String> art
             
             // source root is src/main/java or src/test/java
             File classesDir = tests ? testOutputDirectory : outputDirectory;
-
-            if (!classesDir.exists() && !classesDir.mkdirs()) {
-                throw new PluginExecutionException("The classes output directory " + classesDir.getAbsolutePath() + " does not exist and cannot be created.");
+            if (!classesDir.exists()) {
+                if (!classesDir.mkdirs()) {
+                    throw new PluginExecutionException("The classes output directory " + classesDir.getAbsolutePath()
+                            + " does not exist and cannot be created.");
+                } else {
+                    // redeploy application when class directory has been created
+                    redeployApp();
```

Comment:
redeploy the app only if class dir is created (i.e. outputDirectory) and we don't need to redeploy the app after the first test class is created in testOutputDirectory.

Reply:
Addressed in commit https://github.com/OpenLiberty/ci.common/pull/94/commits/f297b3ddc49faba28d1395a59020ca206bdfa8f9